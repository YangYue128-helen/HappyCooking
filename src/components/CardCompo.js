import React, {Component}from 'react';
import {
    StyleSheet,
    View,
    Text,
    Image,
} from 'react-native';
import {Card, CardItem, Thumbnail, Body, Left, Right, Button, Icon} from "native-base";
import Ionicons from 'react-native-vector-icons/Ionicons';
import requestIdleCallback from "react-native-web/dist/modules/requestIdleCallback";

class CardCompo extends Component {
    render() {
        const images = {
            "1":require('../images/c1.png'),
            "2":require('../images/c2.png'),
            "3":require('../images/c3.png'),
            "4":require('../images/c4.png'),
            "5":require('../images/c5.png'),
            "6":require('../images/c6.png'),
            "7":require('../images/c7.png'),
            "8":require('../images/c8.png'),
        }
        return(
            <Card>
                <CardItem>
                    <Left>
                        <Thumbnail source={require('../images/Me.png')}/>
                        <Body>
                            <Text>Helen</Text>
                            <Text note>Jun 25, 2020</Text>
                        </Body>
                    </Left>
                </CardItem>
                <CardItem cardBody>
                    <Image source={images[this.props.imageSource]}
                    style={{height: 200, width: null, flex: 1}}/>
                </CardItem>
                <CardItem style={{height: 45}}>
                    <Left>
                        <Button transparent>
                            <Icon name='ios-heart' style={{color:'black'}}/>
                        </Button>
                        <Button transparent>
                            <Icon name='ios-chatbubbles' style={{ color:'black'}}/>
                        </Button>
                        <Button transparent>
                            <Icon name='ios-send' style={{color:'black'}}/>
                        </Button>
                    </Left>
                </CardItem>

                <CardItem style={{height:20}}>
                    <Text style={{marginTop:-20}}>{this.props.likes}</Text>
                </CardItem>

                <CardItem>
                    <Body>
                        <Text>
                            <Text style={{fontWeight: 'bold'}}>
                                Helen
                            </Text>
                            Nutrition is defined as the processes by
                            which an animal or plant takes in and utilises
                            food substances. Essential nutrients include
                            protein, carbohydrate, fat, vitamins, minerals
                            and electrolytes. Normally, 85% of daily energy
                            use is from fat and carbohydrates and 15% from protein
                        </Text>
                    </Body>
                </CardItem>
            </Card>
        )
    }
}

export default CardCompo;

const styles = StyleSheet.create({
    container: {
        marginRight: 0,
        flex: 1,
        backgroundColor: '#003300',
        alignItems: 'center',
        justifyContent: 'center',
    },
});