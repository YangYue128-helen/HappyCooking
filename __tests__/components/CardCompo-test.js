import React from 'react';
import CardCompo from "../../src/components/CardCompo";


import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<CardCompo />).toJSON();
    expect(tree).toMatchSnapshot();
});