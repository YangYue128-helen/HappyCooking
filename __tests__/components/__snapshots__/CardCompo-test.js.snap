// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderBottomWidth": 0.5,
      "borderColor": "#ccc",
      "borderLeftWidth": 0.5,
      "borderRadius": 2,
      "borderRightWidth": 0.5,
      "borderTopWidth": 0.5,
      "elevation": 3,
      "flexWrap": "nowrap",
      "marginBottom": 5,
      "marginLeft": 2,
      "marginRight": 2,
      "marginTop": 5,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 1.5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "borderRadius": 28,
            "height": 56,
            "width": 56,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": null,
            "alignSelf": "center",
            "flex": 1,
            "marginLeft": 10,
          }
        }
      >
        <Text>
          Helen
        </Text>
        <Text
          note={true}
        >
          Jun 25, 2020
        </Text>
      </View>
    </View>
  </View>
  <View
    cardBody={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": -5,
        "paddingLeft": -5,
        "paddingRight": -5,
        "paddingTop": -5,
      }
    }
  >
    <Image
      style={
        Object {
          "flex": 1,
          "height": 200,
          "width": null,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        },
        Object {
          "height": 45,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingBottom": null,
            "paddingLeft": null,
            "paddingRight": 15,
            "paddingTop": null,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Text />
      </View>
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingBottom": null,
            "paddingLeft": null,
            "paddingRight": 15,
            "paddingTop": null,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Text />
      </View>
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingBottom": null,
            "paddingLeft": null,
            "paddingRight": 15,
            "paddingTop": null,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        },
        Object {
          "height": 20,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "marginTop": -20,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "stretch",
          "flex": 1,
        }
      }
    >
      <Text>
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Helen
        </Text>
        Nutrition is defined as the processes by which an animal or plant takes in and utilises food substances. Essential nutrients include protein, carbohydrate, fat, vitamins, minerals and electrolytes. Normally, 85% of daily energy use is from fat and carbohydrates and 15% from protein
      </Text>
    </View>
  </View>
</View>
`;
